src/main/java/domen/rideapp/infrastructure/map/GoogleMapsService.java

```diff
@@ -0,0 +1,33 @@
+package domen.rideapp.infrastructure.map;
+
+import domen.rideapp.domain.model.Localization;
+import domen.rideapp.domain.model.RouteEstimate;
+import domen.rideapp.infrastructure.exception.ExternalApiException;
+import domen.rideapp.infrastructure.mapping.MapService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.convert.ConversionService;
+
+import java.util.Optional;
+
+public class GoogleMapsService implements MapService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GoogleMapsService.class);
+    private final GoogleMapsClient client;
+    private final ConversionService conversionService;
+
+    public GoogleMapsService(GoogleMapsClient client, ConversionService conversionService) {
+        this.client = client;
+        this.conversionService = conversionService;
+    }
+
+    public Optional<RouteEstimate> getRouteEstimate(Localization localization) {
+        try {
+            String json = client.fetchDistanceMatrixJson(localization);
+            RouteEstimate estimate = conversionService.convert(json, RouteEstimate.class);
+            return Optional.ofNullable(estimate);
```

Comment:
Ja bym tutaj zwracał Optional.of - to zminimalizuje szukanie problemu gdy getRouteEstimate zwraca nulla i nie wiemy czemu (czyli nie ma bledu ExternalApiException ale z jakiegos powodu konwerter zwraca nulla). Czy mamy jakis powod zeby konwerter zwracal nulla skoro nie bedzie problemu ExternalApiException?